x/superfluid/keeper/intermediary_account.go

```diff
@@ -107,10 +110,76 @@ func (k Keeper) SetIntermediaryAccount(ctx sdk.Context, acc types.SuperfluidInte
 	prefixStore.Set(acc.GetAccAddress(), bz)
 }
 
-func (k Keeper) DeleteIntermediaryAccount(ctx sdk.Context, address sdk.AccAddress) {
+func (k Keeper) DeleteAllEmptyIntermediaryAccounts(ctx sdk.Context) {
```

Comment:
Do we have a direct test for this? I think we are only testing it vicariously via `EndBlocker`

Reply:
imo, we don't need direct test as the only thing `EndBlocker` is doing is calling `DeleteAllEmptyIntermediaryAccounts`. Even if we do have a test case, the test cases would look exactly the same.

I think the current test of `TestDeleteAllEmptyIntermediaryAccounts` should not be seen as a vicorous way of testing, but fully testing the original method + testing endblocker execution

lmk what you think